scoap3/cli_fixes.py

```diff
@@ -252,3 +253,50 @@ def fix_publication_date(dry_run, ids, min_day_diff):
 
     if dry_run:
         error('NO CHANGES were committed to the database, because --dry-run flag was present.')
+
+
+def change_oai_hostname_for_record(record, old_hostname, new_hostname, dry_run):
+    if not record.json:
+        rerror('no json', record)
+        return
+
+    if 'id' not in record.json.get('_oai', {}):
+        rerror('no oai information', record)
+        return
+
+    pid = PersistentIdentifier.query\
+        .filter(PersistentIdentifier.object_uuid == record.id)\
+        .filter(PersistentIdentifier.pid_type == 'oai').one()
+
+    if old_hostname in pid.pid_value:
+        new_pid_value = pid.pid_value.replace(old_hostname, new_hostname)
+
+        rinfo('%s -> %s' % (pid.pid_value, new_pid_value), record)
+
+        if not dry_run:
+            pid.pid_value = new_pid_value
+            record.json['_oai']['id'] = new_pid_value
+            flag_modified(record, 'json')
```

Comment:
Don't you need to commit to the DB here? 
If yes, you might not want to commit at every update otherwise it will slow down a lot

Reply:
We dont as the `process_all_records` function does that for us. 
That function is basically a wrapper that executes the given function on records in bunches and then commits them. The transaction is commited after processing 50 records.